packages/smooth_app/lib/background/background_task_image.dart

```diff
@@ -329,8 +331,22 @@ class BackgroundTaskImage extends BackgroundTaskUpload {
     return croppedPath;
   }
 
-  static String getCroppedPath(final String fullPath) =>
-      '$fullPath.cropped.jpg';
+  static Future<String> getCroppedPath(final String fullPath) async {
+    String croppedPath = '$fullPath.cropped.jpg';
```

Comment:
Reading again and again your code and the existing code, I assume a quick and maintainable solution would be just:
```suggestion
    return getFile('$fullPath.cropped.jpg');
```

Reply:
Where is this `getFile` method from?

Reply:
Comes from `BackgroundTaskUpload`:
```dart
  static Future<File> getFile(String path) async {
    if (Platform.isIOS) {
      final int lastSeparator = path.lastIndexOf('/');
      final String filename = lastSeparator == -1
          ? path
          : path.substring(lastSeparator + 1);
      final Directory directory = await getDirectory();
      path = '${directory.path}/$filename';
    }
    return File(path);
  }
```

Reply:
Ok, I better understand your point.
With the solution of `getFile`, it will always return the new dir on iOS.
But the algorithm is slightly different here: if the path is reachable, then use it. If not, call `getDir()`

Reply:
@g123k iOS is a pain in the neck, and I'm not able to test it. And not that many people dare to work on background tasks.
I cannot say which algorithm is better, yours or mine, but it would be great if the same algorithm was used for "your" issue and "mine", that both answer the question: "how to deal with iOS changing folders?"